### ISSUE TITLE
request to be able to use other chess engines

### ISSUE THREAD

**Issue by H3X0M** [Psyyke/A.C.A.S#11]
Please make it so that it can use other chess engines

**Comment by Hakorr** [comment]
There aren't really any other engines that have been ported to JavaScript that work as good as Stockfish.

**Comment by H3X0M** [comment]
How about komodo engine ?

https://raw.githubusercontent.com/Frosty54/komodo.js/main/src/komodo.js

**Comment by H3X0M** [comment]
How about komodo engine ?

https://raw.githubusercontent.com/Frosty54/komodo.js/main/src/komodo.js

**Comment by kusayuzayushko** [comment]
Please consider adding lc0 and an option. https://github.com/frpays/lc0-js
There are nets for lc0 that ca emulate human play on various elo-levels from 1200 to 1900 
 https://github.com/CSSLab/maia-chess 
There are also fun, super aggressive nets that are super small and don't require much computation https://github.com/dkappe/leela-chess-weights/wiki/Bad-Gyal
And many more https://lczero.org/dev/wiki/best-nets-for-lc0/

**Comment by kusayuzayushko** [comment]
Is there a way to run ACAS on localhost and add local engine like lc0 there? I played around with SF and it's really not much of a use compared to lc0. Let me explain. SF play is weird and it gets even nonesensical on low elo settings as it intentionally makes blunders to limit its strength. This is terrible thing to have for a chess assistant. And overall SF play doesn't make sense for both players, it ruins your playstile and it's obvious for everyone who sees it. 
Carefully adjusted lc0 is indistinguishable from human and if you use it as a co-pylot, it actually helps to improve positional play a lot.  I used to run both and it's absolutely clear that using SF for a chess assistant is a dead end.
Please let's think on what can be done?

**Comment by Hakorr** [comment]
> Please let's think on what can be done?

That's possible locally for sure.

**Comment by kusayuzayushko** [comment]
> > Please let's think on what can be done?
> 
> That's possible locally for sure.

Would be awesome to have an ability to add UCI complained chess engine and an ability to select between them with a local build with some basic UI. Maybe even run multiple engines at the same time with different colors for arrows? Please, consider it at least for future updates?

**Comment by beansofhell** [comment]
Perhaps we can use CuteFish (in order to create some kind of api to UCI)

**Comment by cenunix** [comment]
yeah @Hakorr would it be very complicated to add lc0 for localhosting? i see you made some adjustments to zerofish but how many changes do you think would be needed to actually run lc0?

**Comment by Hakorr** [comment]
> would it be very complicated to add lc0 for localhosting? i see you made some adjustments to zerofish but how many changes do you think would be needed to actually run lc0?

To run the latest model it would probably be easier to run it locally than attempt to port it to WASM. As far as I know it's not tremendously hard to make A.C.A.S listen for some localhost service, just haven't implemented it as I have figured those wanting to use their own engines and mess around locally can probably find another project like A.C.A.S but one that is external and meant for more advanced users.

**Comment by cenunix** [comment]
Ah, I haven't found anything that's similar yet in terms of A.C.A.S and its connection to websites. I think having engines running locally and connecting to A.C.A.S as you described would be great though
